plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -673,6 +744,58 @@ func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes
 	return nil
 }
 
+func (r *ReferenceManager) ensureBackupEntryReferences(ctx context.Context, oldBackupEntry, backupEntry *core.BackupEntry) error {
+	if !equality.Semantic.DeepEqual(oldBackupEntry.Spec.SeedName, backupEntry.Spec.SeedName) {
+		if backupEntry.Spec.SeedName != nil {
+			if _, err := r.seedLister.Get(*backupEntry.Spec.SeedName); err != nil {
+				return err
+			}
+		}
+	}
+
+	if !equality.Semantic.DeepEqual(oldBackupEntry.Spec.BucketName, backupEntry.Spec.BucketName) {
+		if _, err := r.backupBucketLister.Get(backupEntry.Spec.BucketName); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *ReferenceManager) ensureBackupBucketReferences(ctx context.Context, oldBackupBucket, backupBucket *core.BackupBucket, operation admission.Operation) error {
+	if !equality.Semantic.DeepEqual(oldBackupBucket.Spec.SeedName, backupBucket.Spec.SeedName) {
+		if backupBucket.Spec.SeedName != nil {
+			if _, err := r.seedLister.Get(*backupBucket.Spec.SeedName); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := r.lookupSecret(ctx, backupBucket.Spec.SecretRef.Namespace, backupBucket.Spec.SecretRef.Name); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReferenceManager) validateBackupBucketDeletion(ctx context.Context, backupBucketName string) error {
+	backupEntryList, err := r.gardenExternalCoreClient.CoreV1beta1().BackupEntries("").List(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{core.BackupEntryBucketName: backupBucketName}).String(),
+	})
+	if err != nil {
+		return err
+	}
+	associatedBackupEntries := make([]string, 0)
```

Comment:
```suggestion
	associatedBackupEntries := make([]string, 0, len(backupEntryList.Items))
```

Reply:
hmm.. this creates a slice of size n, and when we append m elements to it, the size becomes m+n.

Reply:
It will create a silce of size N, but length 0 so append will add from the start. If it was `make([]string, len(backupEntryList.Items)` (basically without an argument for the length) then it would be as you say .